### ISSUE TITLE
 The AWS Access Key Id you provided does not exist in our records.

### ISSUE THREAD

**Issue by rohanrajpal** [beckn/biab-bap-protocol#15]
On running `./gradlew build` I'm getting the following output.

```bash
biab-bap-protocol on  main via ☕ v11.0.13 via 🅺  took 1m33s
❯ ./gradlew build --scan                                                                                 (base) 18:24:20
> Task :kaptGenerateStubsKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':kaptGenerateStubsKotlin'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not resolve org.beckn.jvm:beckn-protocol-dtos:0.9.3.20.
     Required by:
         project :
      > Could not resolve org.beckn.jvm:beckn-protocol-dtos:0.9.3.20.
         > Could not get resource 's3://beckn-maven-artifacts/releases/org/beckn/jvm/beckn-protocol-dtos/0.9.3.20/beckn-protocol-dtos-0.9.3.20.pom'.
            > The AWS Access Key Id you provided does not exist in our records. (Service: Amazon S3; Status Code: 403; Error Code: InvalidAccessKeyId; Request ID: 6QE2JN03G7C129PF; S3 Extended Request ID: X6pXNtKwx6tRB41L5ofkaAwie8e8LtmQIB6UXBqpk6fws6RiSv2M44uSqCUYP1rx8E/Ruu7B8pQ=; Proxy: null)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed

Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service defined at https://gradle.com/terms-of-service. Do you accept these terms? [yes, no] yes

Gradle Terms of Service accepted.

Publishing build scan...
https://gradle.com/s/ukdet5dox5sng
```

Here's the build scan for detailed info:
https://gradle.com/s/ukdet5dox5sng

**Comment by rohanrajpal** [comment]
@core-wg-admin could you please help me out here?

**Comment by gamemaker1** [comment]
This issue was solved on the slack channel, so I am posting the solution here in case anyways stumbles across it later:

The version of the `protocol-dtos` library that was built and published locally was `0.9.3.26`, and the version required by the `protocol-helper` was `0.9.3.20` - when gradle couldn't find `0.9.3.20` locally, it made a request to the AWS maven repo, which resulted in an auth error.

The solution was to change this line:

https://github.com/beckn/biab-bap-protocol/blob/d75408edd0fd3d5826793f6e3c343dc782220bce/build.gradle.kts#L51

... to:

```gradle
implementation("org.beckn.jvm:beckn-protocol-dtos:0.9.3.+") 
```

> The `+` sign at the end is a wildcard character - which means accept any version that matches the regex `0.9.3.*`

and then rebuild the project using `gradle clean build`.
